<a href="https://colab.research.google.com/github/HerbertAnchovy/notebooks/blob/master/Week_1_project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Project 1: Deaths by tuberculosis

by Michel Wermelinger, 14 July 2015, with minor edits from Jez Phipps on 5<sup>th</sup> April 2016. Notebook revised on 19<sup>th</sup> June 2019 for use on Colaboratory.

This is the project notebook for Week 1 of The Open University's [_Learn to code for Data Analysis_](http://futurelearn.com/courses/learn-to-code) course.

In 2000, the United Nations set eight Millenium Development Goals (MDGs) to reduce poverty and diseases, improve gender equality and environmental sustainability, etc. Each goal is quantified and time-bound, to be achieved by the end of 2015. Goal 6 is to have halted and started reversing the spread of HIV, malaria and tuberculosis (TB).
TB doesn't make headlines like Ebola, SARS (severe acute respiratory syndrome) and other epidemics, but is far deadlier. For more information, see the World Health Organisation (WHO) page <http://www.who.int/gho/tb/en/>.

Given the population and number of deaths due to TB in some countries during one year, the following questions will be answered: 

- What is the total, maximum, minimum and average number of deaths in that year?
- Which countries have the most and the least deaths?
- What is the death rate (deaths per 100,000 inhabitants) for each country?
- Which countries have the lowest and highest death rate?

The death rate allows for a better comparison of countries with widely different population sizes.

## The data

The data consists of total population and total number of deaths due to TB (excluding HIV) in 2013 in each of the BRICS (Brazil, Russia, India, China, South Africa) and Portuguese-speaking countries. 

The data was taken in July 2015 from <http://apps.who.int/gho/data/node.main.POP107?lang=en> (population) and <http://apps.who.int/gho/data/node.main.593?lang=en> (deaths). The uncertainty bounds of the number of deaths were ignored.

The data was collected into an Excel file which should be in the same folder as this notebook.

In [0]:
import warnings

warnings.simplefilter('ignore', FutureWarning)

from pandas import *

from google.colab import files
uploaded = files.upload()

import io
excel_data = io.BytesIO(uploaded['WHO POP TB some.xls'])  
data = read_excel(excel_data)
data

Saving WHO POP TB some.xls to WHO POP TB some (6).xls


Country  Population (1000s)  TB deaths
0                  Angola               21472       6900
1                  Brazil              200362       4400
2                   China             1393337      41000
3       Equatorial Guinea                 757         67
4           Guinea-Bissau                1704       1200
5                   India             1252140     240000
6              Mozambique               25834      18000
7                Portugal               10608        140
8      Russian Federation              142834      17000
9   Sao Tome and Principe                 193         18
10           South Africa               52776      25000
11            Timor-Leste                1133        990

## The range of the problem

The column of interest is the last one.

In [0]:
tbColumn = data['TB deaths']

The total number of deaths in 2013 is:

In [0]:
tbColumn.sum()

354715

The largest and smallest number of deaths in a single country are:

In [0]:
tbColumn.max()

240000

In [0]:
tbColumn.min()

18

From less than 20 to almost a quarter of a million deaths is a huge range. The average number of deaths, over all countries in the data, can give a better idea of the seriousness of the problem in each country.
The average can be computed as the mean or the median. Given the wide range of deaths, the median is probably a more sensible average measure.

In [0]:
tbColumn.mean()

29559.583333333332

In [0]:
tbColumn.median()

5650.0

The median is far lower than the mean. This indicates that some of the countries had a very high number of TB deaths in 2013, pushing the value of the mean up.

## The most affected

To see the most affected countries, the table is sorted in ascending order by the last column, which puts those countries in the last rows.

In [0]:
data.sort_values('TB deaths')

Country  ...  TB deaths (per 100,000)
9   Sao Tome and Principe  ...                 9.326425
3       Equatorial Guinea  ...                 8.850727
7                Portugal  ...                 1.319759
11            Timor-Leste  ...                87.378641
4           Guinea-Bissau  ...                70.422535
1                  Brazil  ...                 2.196025
0                  Angola  ...                32.134873
8      Russian Federation  ...                11.901928
6              Mozambique  ...                69.675621
10           South Africa  ...                47.370017
2                   China  ...                 2.942576
5                   India  ...                19.167186

[12 rows x 4 columns]

The table raises the possibility that a large number of deaths may be partly due to a large population. To compare the countries on an equal footing, the death rate per 100,000 inhabitants is computed.

In [0]:
populationColumn = data['Population (1000s)']
data['TB deaths (per 100,000)'] = tbColumn * 100 / populationColumn
data.sort_values('TB deaths (per 100,000)')

Country  ...  TB deaths (per 100,000)
7                Portugal  ...                 1.319759
1                  Brazil  ...                 2.196025
2                   China  ...                 2.942576
3       Equatorial Guinea  ...                 8.850727
9   Sao Tome and Principe  ...                 9.326425
8      Russian Federation  ...                11.901928
5                   India  ...                19.167186
0                  Angola  ...                32.134873
10           South Africa  ...                47.370017
6              Mozambique  ...                69.675621
4           Guinea-Bissau  ...                70.422535
11            Timor-Leste  ...                87.378641

[12 rows x 4 columns]

## Conclusions

The BRICS and Portuguese-speaking countries had a total of about 350 thousand deaths due to TB in 2013. The median shows that half of these coutries had fewer than 5,650 deaths. The much higher mean (over 29,000) indicates that some countries had a very high number. The least affected were Sao Tome and Principe and Equatorial Guinea, with 18 and 67 deaths respectively, and the most affected were China and India with 41 thousand and 240 thousand deaths in a single year. However, taking the population size into account, the least affected were Portugal and Brazil with less than 2.2 deaths per 100 thousand inhabitants, and the most affected were Guinea-Bissau and East Timor with over 70 deaths per 100,000 inhabitants.

One should not forget that most values are estimates, and that the chosen countries are a small sample of all the world's countries. Nevertheless, they convey the message that TB is still a major cause of fatalities, and that there is a huge disparity between countries, with several ones being highly affected.